### PR TITLE
Add some infra to check out the project and setup basic CI.

### PR DESCRIPTION
**Pull request by stellaraccident**
The CI itself is temporary and just a first step. There are various naming/path issues we need to address, but I would rather do those with a basic CI functioning.

### CODE REVIEW

File: build_tools/sync.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+              ])
+  iree_version, iree_commit = probe_iree_compiler_version()
+  log(f"Using IREE compiler binary version={iree_version} at {iree_commit}")
+  dylib_path = probe_iree_compiler_dylib()
+  log(f"Found IREE compiler dylib: {dylib_path}")
+  iree_path = checkout_repo("iree",
+                            clone_url="https://github.com/iree-org/iree.git",
+                            commit=iree_commit)
+  sync_iree_runtime_submodules(iree_path)
+  jax_path = checkout_repo("jax", "https://github.com/google/jax.git")
```
Comment (jpienaar) [inline build_tools/sync.py:54]: I think there is a also a jaxlib nightly, but if this works then good.

**Review by jpienaar — APPROVED** [review]
Nice, I think this is perfect starting point/everything looks reasonable (I was going to comment on bazel-bin for which I think there is a more robust way to query, but doubt needed in practice/can be addressed only if needed).

File: build_tools/sync.py (hunk 1 above)
Comment (stellaraccident) [reply build_tools/sync.py:54]: Yeah - we're in a weird state right now with pins to jax/tensorflow so going to let it float (I've got some local patches).  Will make better later.

**Comment by stellaraccident** [comment]
I'm going to land this broken since it is highlighting some API drift I have to fix forward.

**Pull request merged**